pkg/providers/vsphere/placement/zone_placement.go

```diff
@@ -112,6 +112,10 @@ func getPlacementCandidates(
 			zones = append(zones, zone)
 		}
 		for _, zone := range zones {
+			// Filter out the zone that is to be deleted, so we don't have it as a candidate.
```

Comment:
I think this ends up being too strict for the `!zonePlacement` case above. I think draining zones just need to be ignored when we are doing zone placement.

Reply:
makes sense. Is it for the scenario where we have only one draining zone as candidate, we don't want to block and saying no candidate available?

Reply:
No. If the VM specified the zone at create time then it passed the webhook and we should try to create it. For when we're otherwise doing zone placement draining zones should not be included in the candidates. I don't want to go down the path of inferring SSO user or not via the OwnerRefs but we might be forced to do so later.